### ISSUE TITLE
ksonnet packages for Pachyderm

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#611]
See #151 for a larger proposal about how Pachyderm might integrate with other Kubeflow components.

This issue is specifically for getting a Kubeflow package for pachyderm (#610)

I started with the [helm chart](https://github.com/kubernetes/charts/tree/master/stable/pachyderm/templates)

But it looks like that is missing a bunch of components e.g.
- storage class
- dashboard

It looks like some of these resources are created in code e.g. in [assets.go](
https://github.com/pachyderm/pachyderm/blob/master/src/server/pkg/deploy/assets/assets.go)

@dwhitena @jdoliner @JoeyZwicker Could you provide me with a set of YAML/json manifests for all the K8s resources that need to be created as part of a pachyderm deployment?

**Comment by jlewi** [comment]
So I was able to connect by doing

```
pachctl --namespace=kubeflow port-forward
```

Didn't realize the dashboard was part of the enterprise edition.

**Comment by pdmack** [comment]
Yeah, was wondering that. Seems like just the CLI for the community version.

**Comment by JoeyZwicker** [comment]
The dashboard is part of the enterprise edition, but if you put in your email it'll give you a free 2-week trial token.

**Comment by jlewi** [comment]
@JoeyZwicker Thanks but I don't have an immediate need to use the dashboard; now that I know its not part of the community edition I removed it from the ksonnet package.

I deployed pachyderm using the CLI configured to use GCS and here is a list of the K8s resources I see created

```
serviceaccount "pachyderm" configured
clusterrole "pachyderm" configured
clusterrolebinding "pachyderm" configured
storageclass "etcd-storage-class" configured
service "etcd-headless" configured
statefulset "etcd" configured
service "etcd" configured
service "pachd" created
deployment "pachd" configured
service "dash" configured
deployment "dash" configured
secret "pachyderm-storage-secret" configured
```

**Comment by jlewi** [comment]
Looks like its using cluster roles where as the helm chart was just using roles.

Here is a list of all the K8s resources created when I deployed via the CLI and configured to use GCS.
[pachyd_resources.txt](https://github.com/kubeflow/kubeflow/files/1892006/pachyd_resources.txt)

**Comment by jlewi** [comment]
@dwhitena @JoeyZwicker it looks like etcd is running as a statefulset in the GCP case but as a deployment in the non GCP case? 

Any reason its not always running as a statefulset?

**Comment by jdoliner** [comment]
@jlewi when stateful sets first came out we setup etcd to work with stateful sets but we found that GCP was the only place that really had functioning support for them. I suspect that's improved at this point some but we still do get a number of issues where people deploy stateful sets on AWS and EBS volumes get stuck and things like that. A couple of times we've had to get people to redeploy w/o stateful sets to get things working on other platforms. So it's mostly just an artifact of what worked and what didn't and tweaking the defaults to route people around common support issues. I'm not totally sure how current this information is though, I don't quite remember when exactly the last time someone had an issue like this was.
